### DISCUSSION TITLE
bag file

### DISCUSSION THREAD

**Discussion by prajyotkoli123** [community/community#21368] [Other Feature Feedback, Questions, & Ideas]
<p>I want to parse bag file using java and convert it into csv file.</p>
<p>Please help.</p>

**Comment by fire-eggs** [comment]
<p>Could you be a little more specific?  I.e.</p>
<ol>
<li>Do you need to convert .bag files to .csv and don’t care how,</li>
</ol>
<p>or </p>
<ol start="2">
<li>You have a specific need to use Java to parse .bag files and write .csv</li>
</ol>
<p>?</p>

**Comment by prajyotkoli123** [comment]
<p>Yes. I have specific need to use java to parse .bag files and write .csv</p>

**Comment by fire-eggs** [comment]
<p>How about <a href="https://github.com/swri-robotics/bag-reader-java" rel="nofollow noopener">this</a> ?</p>
<p>Otherwise, most parsers seem to be C/C++, where you’d have to translate.</p>

**Comment by prajyotkoli123** [comment]
<p>Thanks for replay.</p>
<p>I refered this link but suppose if I have random bag file and i don’t know which type of data present on topic present then how do I parse the bag file.</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>You’ll have to follow a multi-pass process. As it shows in the readme file for the repository, you can print all the topics in the bagfile. Once you know the topics, you can extend your program to grab the topics of interest.</p>
<p>Or write a picker where you list the available topics, and pick from those topics to view more details.</p>
<p>Only you know what information you want to pull out of the bagfile; if there is no “standard” to help you identify the information you want, then you’re going to have to use your own judgement.</p>

**Comment by prajyotkoli123** [comment]
<p>Thanks for help. Yes I need to go through topic information.</p>
